Submitted By: Ken Moffat <ken at linuxfromscratch dot org>
Date: 2018-05-06
Initial Package Version: 0.4.0
Upstream Status: Unknown
Origin: Found at Arch, rediffed to apply with -p1.
Description: Fixes build if ffmpeg-0.4.0 has been installed.
The existing code only builds with ffmpeg-3.4 and earlier.

diff -Naur gegl-0.4.0/operations/external/ff-save.c gegl-0.4.0.patched/operations/external/ff-save.c
--- gegl-0.4.0/operations/external/ff-save.c	2018-04-24 17:42:13.000000000 +0100
+++ gegl-0.4.0.patched/operations/external/ff-save.c	2018-05-04 22:29:16.025521327 +0100
@@ -290,7 +290,7 @@
   c->codec_type = AVMEDIA_TYPE_AUDIO;
 
   if (oc->oformat->flags & AVFMT_GLOBALHEADER)
-    c->flags |= CODEC_FLAG_GLOBAL_HEADER;
+    c->flags |= AV_CODEC_FLAG_GLOBAL_HEADER;
 
   return st;
 }
@@ -631,7 +631,7 @@
 #endif
 
    if (oc->oformat->flags & AVFMT_GLOBALHEADER)
-     c->flags |= CODEC_FLAG_GLOBAL_HEADER;
+     c->flags |= AV_CODEC_FLAG_GLOBAL_HEADER;
 
   return st;
 }
@@ -699,7 +699,9 @@
     }
 
   p->video_outbuf = NULL;
+#ifdef AVFMT_RAWPICTURE
   if (!(oc->oformat->flags & AVFMT_RAWPICTURE))
+#endif
     {
       /* allocate output buffer, 1 mb / frame, might fail for some codecs on UHD - but works for now */
       p->video_outbuf_size = 1024 * 1024;
@@ -803,6 +805,7 @@
   picture_ptr      = p->picture;
   picture_ptr->pts = p->frame_count;
 
+#ifdef AVFMT_RAWPICTURE
   if (oc->oformat->flags & AVFMT_RAWPICTURE)
     {
       /* raw video case. The API will change slightly in the near
@@ -821,6 +824,7 @@
       ret = av_write_frame (oc, &pkt);
     }
   else
+#endif
     {
       /* encode the image */
       AVPacket pkt2;
